dev_container

```diff
@@ -712,7 +712,7 @@ launch() {
         ${ucx_opt} \
         ${docker_opts} \
         ${img} \
-        "$@"
+        -c "$@"
```

Comment:
On review, I'd request that we maintain the existing "$@" behavior. Reasoning:
- The proposed behavior introduces a regression where `dev_container launch` no longer launches into an interactive shell and instead exits immediately.
- `-c` itself is a forwarded argument rather than a part of the `docker run` command. Under typical execution the trailing arguments here are forwarded first to the `/opt/nvidia/nvidia_entrypoint.sh` container script and then on to the bash `exec` function. It makes sense that an outer call to `dev_container launch` should include `-c` among a larger list of arguments to forward to the container.
- On inspection `-c` itself doesn't seem to be required to run a forwarded command with `exec`. If you're seeing issues with multi-arg forwarding, maybe the cause lies elsewhere?

Based the bash [manpage](https://manpages.ubuntu.com/manpages/trusty/en/man1/bash.1.html) I don't anticipate we have a strong use case for alternative arguments to `exec`, but it would be good to avoid changing default `dev_container launch` interactive behavior. Are you seeing in your testing that multi-arg forwarding fails without this update?

Reply:
@tbirdso good catch for the `dev_container launch` issue, thank you!  

I've reverted that change and moved it inside `dev_container build_and_run`. 
Yes, some change is needed in order to get multi-arg to be forwarded correctly after recent changes to how the args were being forwarded.  The current proposed changes in dev_container now seem to work both interactively and with multi-arg forwarding.

Reply:
Verified the update works on my system, thanks!